propagation.md

```diff
@@ -1,73 +1,74 @@
+Any SpanContext in a trace may be **Injected** into what OpenTracing refers to as a Carrier. A **Carrier** is an interface or data structure that's *useful* for inter-process communication (IPC); that is, the Carrier is something that "carries" the tracing state from one process to another. The OpenTracing specification includes two [required Carrier formats](#required-carriers), though [custom Carrier formats](#custom-carriers) are possible as well.
 
-Similarly, given a Carrier, an injected trace may be **Joined**, yielding a new Span. We use the term "Join" rather than "Extract" (or similar) because Inject/Join is not a symmetric process: an OpenTracing implementation is only expected to inject the minimal amount of information needed to constunct a (descendant) Span in the peer and join it to the overarching trace. The injected state is *not* necessarily sufficient to reconstruct the entire injected Span instance, symmetrically.
+Similarly, given a Carrier, an injected trace may be **Extracted**, yielding a SpanContext instance which is semantically identical to the one Injected into the Carrier.
 
 #### Inject pseudocode example
 
 {% highlight python %}
-span = ...
-outboundRequest = ...
+span_context = ...
+outbound_request = ...
 
 # We'll use the (builtin) TEXT_MAP carrier format. We start by
 # using an empty map as the carrier prior to the call to
 # `tracer.inject`.
 carrier = {}
-tracer.inject(span, opentracing.Format.TEXT_MAP, carrier)
+tracer.inject(span_context, opentracing.Format.TEXT_MAP, carrier)
 
 # `carrier` now contains (opaque) key:value pairs which we pass
 # along over whatever wire protocol we already use.
 for key, value in carrier:
-    outboundRequest.headers[key] = escape(value)
+    outbound_request.headers[key] = escape(value)
 {% endhighlight %}
 
-#### Join pseudocode example
+#### Extract pseudocode example
 
 {% highlight python %}
-inboundRequest = ...
+inbound_request = ...
 
 # We'll again use the (builtin) TEXT_MAP carrier format. Per the
 # TEXT_MAP documentation, we can use a map that has extraneous data
 # in it and let the OpenTracing implementation look for the subset
 # of key:value pairs it needs.
 #
-# As such, we directly use the key:value `inboundRequest.headers`
+# As such, we directly use the key:value `inbound_request.headers`
 # map as the carrier.
-carrier = inboundRequest.headers
-# The operation name we want for the server-side span:
-operationName = ...
-serverSpan = tracer.join(
-    operationName, opentracing.Format.TEXT_MAP, carrier)
+carrier = inbound_request.headers
+span_context = tracer.extract(opentracing.Format.TEXT_MAP, carrier)
+// Continue the trace given span_context. E.g.,
+//
+//     span = tracer.start_span("...", child_of=span_context)
+//
```

Comment:
(good catch)
ᐧ

On Sat, Jul 2, 2016 at 2:44 PM, Yuri Shkuro <notifications@github.com>
wrote:

> In propagation.md
> <https://github.com/opentracing/opentracing.github.io/pull/100#discussion_r69383070>
> :
>
> >  # map as the carrier.
> > -carrier = inboundRequest.headers
> > -# The operation name we want for the server-side span:
> > -operationName = ...
> > -serverSpan = tracer.join(
> > -    operationName, opentracing.Format.TEXT_MAP, carrier)
> > +carrier = inbound_request.headers
> > +span_context = tracer.extract(opentracing.Format.TEXT_MAP, carrier)
> > +// Continue the trace given span_context. E.g.,
> > +//
> > +//     span = tracer.start_span("...", child_of=span_context)
> > +//
>
> Python comments start with # line 70 does not need to be commented out
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/opentracing/opentracing.github.io/pull/100/files/8f9f55ae288e97dfee2eb7c3f92314dfdc2906ff#r69383070>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACJbIcmiodn7zq6Esu6pSLPzmBNfaIvNks5qRtvRgaJpZM4I-cTo>
> .
>